src/common/webview.cpp

```diff
@@ -51,6 +51,154 @@ wxDEFINE_EVENT( wxEVT_WEBVIEW_TITLE_CHANGED, wxWebViewEvent );
 
 wxStringWebViewFactoryMap wxWebView::m_factoryMap;
 
+wxWebViewZoom wxWebView::GetZoom() const
+{
+    float zoom = GetZoomFactor();
+
+    // arbitrary way to map float zoom to our common zoom enum
+    if (zoom <= 0.55)
+    {
+        return wxWEBVIEW_ZOOM_TINY;
+    }
+    else if (zoom > 0.55 && zoom <= 0.85)
+    {
+        return wxWEBVIEW_ZOOM_SMALL;
+    }
+    else if (zoom > 0.85 && zoom <= 1.15)
+    {
+        return wxWEBVIEW_ZOOM_MEDIUM;
+    }
+    else if (zoom > 1.15 && zoom <= 1.45)
+    {
+        return wxWEBVIEW_ZOOM_LARGE;
+    }
+    else if (zoom > 1.45)
+    {
+        return wxWEBVIEW_ZOOM_LARGEST;
+    }
+
+    // to shut up compilers, this can never be reached logically
+    wxASSERT(false);
+    return wxWEBVIEW_ZOOM_MEDIUM;
+}
+
+void wxWebView::SetZoom(wxWebViewZoom zoom)
+{
+    // arbitrary way to map our common zoom enum to float zoom
+    switch (zoom)
+    {
+        case wxWEBVIEW_ZOOM_TINY:
+            SetZoomFactor(0.4f);
+            break;
+
+        case wxWEBVIEW_ZOOM_SMALL:
+            SetZoomFactor(0.7f);
+            break;
+
+        case wxWEBVIEW_ZOOM_MEDIUM:
+            SetZoomFactor(1.0f);
+            break;
+
+        case wxWEBVIEW_ZOOM_LARGE:
+            SetZoomFactor(1.3f);
+            break;
+
+        case wxWEBVIEW_ZOOM_LARGEST:
+            SetZoomFactor(1.6f);
+            break;
+
+        default:
+            wxASSERT(false);
+    }
+}
+
+bool wxWebView::QueryCommandEnabled(const wxString& command) const
+{
+    wxString resultStr;
+    const_cast<wxWebView*>(this)->RunScript(
```

Comment:
Maybe we should make `RunScript()` const? Having these const casts everywhere is not much fun...